compiler/src/main/kotlin/visitors/valueextractor.kt

```diff
@@ -152,4 +153,54 @@ class LiteralExtractorVisitor : BaseASTVisitor() {
         super.visit(node)
         node.ident = node.ctx.var_ident().text
     }
+
+    /**
+     * Parse a string into a integer, or throw IntegerParisingException the string isn't a valid integer.
+     *
+     * @throws IntegerParsingException
+     */
+    private fun parseInt(text: String, node: AST): Int {
+        try {
+            return text.toInt()
+        } catch (e: NumberFormatException) {
+            throw IntegerParsingException(text, node)
+        }
+    }
+
+    override fun visit(node: WorldNode) {
+        /**
+         * Parses a single dimension from the world size declaration list
+         */
+        fun parseDimension(dim: CellmataParser.World_size_dimContext): WorldDimension {
+            val type = dim.type
+            return WorldDimension(
+                size = parseInt(dim.size.text, node),
+                type = when(type) {
+                    is CellmataParser.DimFiniteEdgeContext -> WorldType.EDGE
+                    is CellmataParser.DimFiniteWrappingContext -> WorldType.WRAPPING
+                    else -> throw AssertionError()
```

Comment:
Should this not utilise our newly created ErrorLogger?

Reply:
I don't think so, this is a case where if we hit the else statements it is not the user's fault, it's because the compiler is broken.

Reply:
Okay, cool